.gitlab-ci.yml

```diff
@@ -72,22 +81,24 @@ push_build_environment:
       - master
   stage: push
   script:
-    - docker login -u gitlab-ci-token -p "${CI_JOB_TOKEN}" "${CI_REGISTRY}"
-    - docker tag  "${CI_COMMIT_SHA}:${CI_PIPELINE_ID}" "${CI_REGISTRY_IMAGE}:${CI_COMMIT_SHA}"
-    - docker push "${CI_REGISTRY_IMAGE}:${CI_COMMIT_SHA}"
-    - docker tag  "${CI_REGISTRY_IMAGE}:${CI_COMMIT_SHA}" "${CI_REGISTRY_IMAGE}:latest"
-    - docker push "${CI_REGISTRY_IMAGE}:latest"
+    - |
+      docker login -u gitlab-ci-token -p "${CI_JOB_TOKEN}" "${CI_REGISTRY}"
+      docker tag  "${CI_REGISTRY_IMAGE}:${CI_COMMIT_SHA}" "${CI_REGISTRY_IMAGE}:latest"
+      docker push "${CI_REGISTRY_IMAGE}:latest"
 
 cleanup_build_environment:
   <<: *build_environment_common
-  only:
-    <<: *changes_docker
   stage: cleanup
+  except:
+    refs:
+      - master
```

Comment:
Don't cleanup when we're changing the master Docker image to match the old behavior of storing old images.